staging/src/k8s.io/apiserver/pkg/cel/environment/environment.go

```diff
@@ -231,13 +239,26 @@ func (e *EnvSet) Extend(options ...VersionedOptions) (*EnvSet, error) {
 	return e, nil
 }
 
-func (e *EnvSet) filterAndBuildOpts(base *cel.Env, compatVer *version.Version, opts []VersionedOptions) (cel.EnvOption, error) {
+func (e *EnvSet) filterAndBuildOpts(base *cel.Env, compatVer *version.Version, honorFeatureGateEnablement bool, opts []VersionedOptions) (cel.EnvOption, error) {
 	var envOpts []cel.EnvOption
 	var progOpts []cel.ProgramOption
 	var declTypes []*apiservercel.DeclType
 
 	for _, opt := range opts {
+		var allowedByFeatureGate, allowedByVersion bool
```

Comment:
```suggestion
		allowedByFeatureGate := true
		allowedByVersion := false
```

The variables were not being added to `NewExpressions` for `mustBuildEnvs` when `opt.FeatureEnabled == nil` which set `allowedByFeatureGate` to `false`.

Reply:
Actually, nevermind, this isn't quite enough.

Reply:
> The variables were not being added to NewExpressions for mustBuildEnvs when opt.FeatureEnabled == nil (which was causing allowedByFeatureGate to remain false).

but that falls back to IntroducedVersion / RemovedVersion logic, which is what I want... 

also, the thing failing the test is not NewExpressions, the VAP API write and validation persists fine... it's the compilation by the VAP *admission* plugin (which I *hope* is using the stored expressions env) that was failing to resolve the root variables